scripts/tests/unit/test_schema_cleaner.py

```diff
@@ -0,0 +1,271 @@
+import os
```

Comment:
I didn't see the new unit tests in `./scripts/tests/unit` picked up by `make test` until I added `__init__.py` file to the `unit` testing directory. Are you seeing the same?

<details>
<summary>Before</summary>

```
$ make test
build/ve/bin/python -m unittest discover -v --start-directory scripts/tests
test_clean_string_values (test_ecs_helpers.TestECSHelpers) ... ok
test_dict_copy_existing_keys (test_ecs_helpers.TestECSHelpers) ... ok
test_dict_copy_existing_keys_overwrites (test_ecs_helpers.TestECSHelpers) ... ok
test_get_tree_by_ref (test_ecs_helpers.TestECSHelpers) ... ok
test_is_intermediate_field (test_ecs_helpers.TestECSHelpers) ... ok
test_list_subtract (test_ecs_helpers.TestECSHelpers) ... ok
test_merge_dicts (test_ecs_helpers.TestECSHelpers) ... ok
test_merge_dicts_raises_if_duplicate_key_added (test_ecs_helpers.TestECSHelpers) ... ok
test_sorted_by_multiple_keys (test_ecs_helpers.TestECSHelpers) ... ok
test_sorted_by_one_key (test_ecs_helpers.TestECSHelpers) ... ok
test_base_flat_name (test_ecs_spec.TestEcsSpec) ... ok
test_flat_includes_reusable_fields (test_ecs_spec.TestEcsSpec) ... ok
test_nested_includes_reusable_fields (test_ecs_spec.TestEcsSpec) ... ok
test_related_fields_always_arrays (test_ecs_spec.TestEcsSpec) ... ok
test_add_not_nested (test_es_template.TestGeneratorsEsTemplate) ... ok
test_add_siblings (test_es_template.TestGeneratorsEsTemplate) ... ok
test_dict_add_nested (test_es_template.TestGeneratorsEsTemplate) ... ok
test_dict_add_nested_shouldnt_clobber_with_objects (test_es_template.TestGeneratorsEsTemplate) ... ok
test_dict_add_nested_to_explicit_object (test_es_template.TestGeneratorsEsTemplate) ... ok

----------------------------------------------------------------------
Ran 19 tests in 0.502s

OK
```


</details>

<details>
<summary>After</summary>

```
$ make test
build/ve/bin/python -m unittest discover -v --start-directory scripts/tests
test_clean_string_values (test_ecs_helpers.TestECSHelpers) ... ok
test_dict_copy_existing_keys (test_ecs_helpers.TestECSHelpers) ... ok
test_dict_copy_existing_keys_overwrites (test_ecs_helpers.TestECSHelpers) ... ok
test_get_tree_by_ref (test_ecs_helpers.TestECSHelpers) ... ok
test_is_intermediate_field (test_ecs_helpers.TestECSHelpers) ... ok
test_list_subtract (test_ecs_helpers.TestECSHelpers) ... ok
test_merge_dicts (test_ecs_helpers.TestECSHelpers) ... ok
test_merge_dicts_raises_if_duplicate_key_added (test_ecs_helpers.TestECSHelpers) ... ok
test_sorted_by_multiple_keys (test_ecs_helpers.TestECSHelpers) ... ok
test_sorted_by_one_key (test_ecs_helpers.TestECSHelpers) ... ok
test_base_flat_name (test_ecs_spec.TestEcsSpec) ... ok
test_flat_includes_reusable_fields (test_ecs_spec.TestEcsSpec) ... ok
test_nested_includes_reusable_fields (test_ecs_spec.TestEcsSpec) ... ok
test_related_fields_always_arrays (test_ecs_spec.TestEcsSpec) ... ok
test_add_not_nested (test_es_template.TestGeneratorsEsTemplate) ... ok
test_add_siblings (test_es_template.TestGeneratorsEsTemplate) ... ok
test_dict_add_nested (test_es_template.TestGeneratorsEsTemplate) ... ok
test_dict_add_nested_shouldnt_clobber_with_objects (test_es_template.TestGeneratorsEsTemplate) ... ok
test_dict_add_nested_to_explicit_object (test_es_template.TestGeneratorsEsTemplate) ... ok
test_already_normalized_reuse_notation (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_clean (unit.test_schema_cleaner.TestSchemaCleaner)
A high level sanity test ... ok
test_field_defaults (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_field_defaults_dont_override (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_field_raises_on_missing_required_attributes (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_field_simple_cleanup (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_multi_field_defaults_and_precalc (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_multiline_short_description_raises (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_normalize_reuse_notation (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_reusable_schema_raises_on_missing_reuse_attributes (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_schema_cleanup_setting_defaults (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_schema_raises_on_missing_required_attributes (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_schema_simple_cleanup (unit.test_schema_cleaner.TestSchemaCleaner) ... ok
test_calculate_final_values (unit.test_schema_finalizer.TestSchemaFinalizer) ... ok
test_field_group_at_path_find_nested_destination (unit.test_schema_finalizer.TestSchemaFinalizer) ... ok
test_field_group_at_path_for_leaf_object_field_creates_the_section (unit.test_schema_finalizer.TestSchemaFinalizer) ... ok
test_field_group_at_path_leaf_field_not_field_group (unit.test_schema_finalizer.TestSchemaFinalizer) ... ok
test_field_group_at_path_missing_nested_path (unit.test_schema_finalizer.TestSchemaFinalizer) ... ok
test_field_group_at_path_root_destination (unit.test_schema_finalizer.TestSchemaFinalizer) ... ok
test_perform_reuse_with_foreign_reuse_and_self_reuse (unit.test_schema_finalizer.TestSchemaFinalizer) ... ok
test_deep_nesting_representation (unit.test_schema_loader.TestSchemaLoader) ... ok
test_fields_with_subfields_mergeable (unit.test_schema_loader.TestSchemaLoader) ... ok
test_load_schemas_allows_unique_fieldsets (unit.test_schema_loader.TestSchemaLoader) ... ok
test_load_schemas_fail_on_accidental_fieldset_redefinition (unit.test_schema_loader.TestSchemaLoader) ... ok
test_load_schemas_no_custom (unit.test_schema_loader.TestSchemaLoader) ... ok
test_merge_array_attributes (unit.test_schema_loader.TestSchemaLoader) ... ok
test_merge_field_within_schema (unit.test_schema_loader.TestSchemaLoader) ... ok
test_merge_new_schema (unit.test_schema_loader.TestSchemaLoader) ... ok
test_merge_non_array_attributes (unit.test_schema_loader.TestSchemaLoader) ... ok
test_nest_fields (unit.test_schema_loader.TestSchemaLoader) ... ok
test_nest_fields_multiple_intermediate_fields (unit.test_schema_loader.TestSchemaLoader) ... ok
test_nest_fields_recognizes_explicitly_defined_object_fields (unit.test_schema_loader.TestSchemaLoader) ... ok
test_nest_schema_raises_on_missing_schema_name (unit.test_schema_loader.TestSchemaLoader) ... ok
test_basic_merging (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok
test_eval_globs (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok
test_extract_matching_fields_explicit_all_fields_notation (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok
test_extract_matching_fields_shorthand_notation (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok
test_extract_matching_fields_subfields_only_notation (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok
test_extract_matching_individual_field (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok
test_load_subset_definitions_raises_when_no_subset_found (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok
test_merging_superset (unit.test_schema_subset_filter.TestSchemaSubsetFilter) ... ok

----------------------------------------------------------------------
Ran 60 tests in 0.984s

OK
```

</details>

Reply:
Oh jeez good catch. Such a rubyist's mistake 😂 

I kept running the tests one file at a time when developing, so I never noticed that